src/sdk/service/network-validation-service/network-validation-service.ts

```diff
@@ -0,0 +1,475 @@
+          return false;
+        }
+      }
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Takes a set of IP addresses and a set of IP ranges, and indicates whether
+   * any of the IP addresses belong to any of the ranges.
+   * @param {string[]} ips: array of IP strings
+   * @param {Range[]} ranges: array of range objects
+   * @return {boolean} boolean value
+   */
+  isAnyIpInAnyRange(ips: string[], ranges: Range[]): boolean {
+    for (let i = 0; i < ranges.length; i++) {
+      if (this.isAnyIpInRange(ips, ranges[i])) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Indicates whether any IP in an array of IPs is within an IP range.
+   * @param {string[]} ips IP address strings.
+   * @param {Range} range IP range object.
+   * @return {boolean} boolean value
+   */
+  isAnyIpInRange(ips: string[], range: Range): boolean {
+    return ips.filter((ip) => {
+      return this.isInRange(ip, range);
+    }).length > 0;
+  }
+
+  /**
+   * Converts integer to ip address representation.
+   * @param {number} ipInt: Integer to be converted.
+   * @return {string} IP Address string.
+   */
+  getIntegerAsIp(ipInt: number): string {
+    const components = [];
+    const template = 255;
+    for (let i = 3; i >= 0; i--) {
+      const modTemplate = (template << (i * 8));
+      components.push((modTemplate & ipInt) >>> (i * 8));
+    }
+    return components.join('.');
+  }
+
+  /**
+   * Parses string to int then checks if test number is between them.
+   * @param {string} a: first number for comparison.
+   * @param {string} b: second number for comparison.
+   * @param {string} test: number being compared.
+   * @return {boolean}
+   */
+  inRange(a: string, b: string, test: string): boolean {
+    const aInt = parseInt(a, 10);
+    const bInt = parseInt(b, 10);
+    const testInt = parseInt(test, 10);
+    const min = aInt < bInt ? aInt : bInt;
+    const max = aInt > bInt ? aInt : bInt;
+    return (testInt >= min && testInt <= max);
+  }
+
+  /**
+   * Checks if integer representing port is valid.
+   * @param {number} port: port number being checked.
+   * @return {boolean}
+   */
+  isValidPort(port: number): boolean {
+    return (port >= 1 && port <= 65535);
+  }
+
+  /**
+   * Gets the number of IP addresses in a given range. Although unlikely to be
+   * an issue, note that due to the fact that javascript numbers are represented
+   * as signed, two's complement integers, for a network with more than 2^31
+   * addresses this function will return a negative number (May not even be
+   * technically possible). Returns -1 if the parameters are invalid.
+   * @param {Range} ipRange object representing IP range of the form { start:
+   *          "x.x.x.x", end: "x.x.x.x" }
+   * @return {integer} size of range.
+   */
+  getNumberOfAddressesInRange(range: Range): number | null {
+    if (!(this.isValid(range.start) && this.isValid(range.end))) {
+      return null;
```

Comment:
I based this on the network-validation-service in the console code, and it returned `null`. @cfsnyder, what do you think?

Reply:
answered by previous comments. namely, that input should be an IPv4Address type so that you aren't dealing with validation of params within each utility method.